### DISCUSSION TITLE
Completion support for Carapace and/or Nushell

### DISCUSSION THREAD

**Discussion by BatmanAoD** [jj-vcs/jj#2497] [Ideas]
I'm trying to get used to [Nushell](https://github.com/nushell/nushell/) as my daily driver, and as part of that I've configured [Carapace](https://github.com/rsteube/carapace-bin) for most of my completions.

I'll be honest: I know nothing about how to create completions for Carapace or natively in Nushell. But would anyone be interested in trying to build those?

**Comment by BatmanAoD** [comment]
I've also made this as a request in the Carapace repo: https://github.com/rsteube/carapace-bin/issues/1968

**Comment by ilyagr** [comment]
Related PR: https://github.com/martinvonz/jj/pull/2353. If you comment there, it will ping the author.

**Comment by BatmanAoD** [comment]
Carapace maintainers have opened an issue and are already making progress: https://github.com/rsteube/carapace-bin/issues/1970

**Reply by ilyagr** [reply]
It would be interesting if they could support better completion than clap can, e.g. complete branch names.

**Reply by necauqua** [reply]
> It would be interesting if they could support better completion than clap can, e.g. complete branch names.

yeah, they literally already do :)
I've been dying waiting for dynamic clap completion, and they just casually implement it all from outside :)
